### ISSUE TITLE
Crash while running rubocop on an empty file w/ a `-` character in the name.

### ISSUE THREAD

**Issue by bwinter** [bbatsov/rubocop#2910]
## Expected behavior
I am running rubocop on an empty file that has a `-` character in it's name. I expect this to pass all cops w/o any complaint (or at least not crashing).
e.g.
```sh
`--> rubocop thing-a.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## Actual behavior
The actual behavior is to crash
```sh
`--> rubocop config/deploy/testing-a.rb -d
For /home/bwinter/gametimesf/workspace/code: configuration from /home/bwinter/gametimesf/workspace/code/.rubocop.yml
Inheriting configuration from /home/bwinter/gametimesf/workspace/code/.rubocop_todo.yml
Default configuration from /home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/config/default.yml
Inheriting configuration from /home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/config/enabled.yml
Inheriting configuration from /home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/config/disabled.yml
Inspecting 1 file
Scanning /home/bwinter/gametimesf/workspace/code/config/deploy/testing-a.rb
An error occurred while Style/FileName cop was inspecting /home/bwinter/gametimesf/workspace/code/config/deploy/testing-a.rb.
undefined method `start_with?' for nil:NilClass
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/cop/style/file_name.rb:52:in `shebang?'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/cop/style/file_name.rb:36:in `investigate'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/cop/commissioner.rb:91:in `block (2 levels) in invoke_custom_processing'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/cop/commissioner.rb:97:in `with_cop_error_handling'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/cop/commissioner.rb:90:in `block in invoke_custom_processing'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/cop/commissioner.rb:87:in `each'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/cop/commissioner.rb:87:in `invoke_custom_processing'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/cop/commissioner.rb:57:in `investigate'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/cop/team.rb:65:in `inspect_file'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/runner.rb:194:in `inspect_file'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/runner.rb:164:in `block in do_inspection_loop'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/runner.rb:158:in `loop'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/runner.rb:158:in `do_inspection_loop'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/runner.rb:87:in `process_file'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/runner.rb:59:in `block in inspect_files'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/runner.rb:57:in `each'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/runner.rb:57:in `inspect_files'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/runner.rb:35:in `run'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/lib/rubocop/cli.rb:30:in `run'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/bin/rubocop:14:in `block in <top (required)>'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
/home/bwinter/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-0.37.2/bin/rubocop:13:in `<top (required)>'
/home/bwinter/.rbenv/versions/2.2.3/bin/rubocop:23:in `load'
/home/bwinter/.rbenv/versions/2.2.3/bin/rubocop:23:in `<main>'
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Style/FileName cop was inspecting /home/bwinter/gametimesf/workspace//config/deploy/testing-a.rb.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
Mention the following information in the issue report:
0.37.2 (using Parser 2.3.0.6, running on ruby 2.2.3 x86_64-linux)
Finished in 0.10424702987074852 seconds
```

## Steps to reproduce the problem
Create an empty file with a `-` character in the name:
```sh
`--> touch thing-a.rb
```
Then try to run rubocop on the file:
```sh
`--> rubocop thing-a.rb
Inspecting 1 file
An error occurred while Style/FileName cop was inspecting /home/bwinter/thing-a.rb.
To see the complete backtrace run rubocop -d.
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Style/FileName cop was inspecting /home/bwinter/thing-a.rb.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
Mention the following information in the issue report:
0.37.2 (using Parser 2.3.0.6, running on ruby 2.2.3 x86_64-linux)
```

## RuboCop version
Running 0.37.2

```sh
`--> rubocop -V
0.37.2 (using Parser 2.3.0.6, running on ruby 2.2.3 x86_64-linux)
```

**Comment by alexdowad** [comment]
Can't reproduce on ruby 2.2.2 or 2.3.0. Please post your `.rubocop.yml`.

**Comment by bwinter** [comment]
```sh
`--> ruby -v
ruby 2.2.3p173 (2015-08-18 revision 51636) [x86_64-linux]
```

```yml
inherit_from: .rubocop_todo.yml

Style/FileName:
  Exclude:
    - 'config/deploy/*.rb'
```

Hmm, I wonder if it's related to my OS.

**Comment by alexdowad** [comment]
This was fixed in 333ee950, which was *after* 0.37.2. The fix will be there in the next release.

Let's close this issue.
